File: routes/bookRoutes.js
// routes/bookRoutes.js
const express = require('express');
const router = express.Router();
const bookController = require('../controllers/bookController');

const { verifyToken, isAdmin } = require('../middleware/authMiddleware');

router.post('/', verifyToken, isAdmin, bookController.addBook);
// ...

// 图书列表 & 搜索  routes/bookRoutes.js
router.get('/', bookController.getAllBooks);

// 获取单本图书 routes/bookRoutes.js
router.get('/:id', bookController.getBookById);

// 添加图书 routes/bookRoutes.js
router.post('/', bookController.addBook);

// 更新图书 routes/bookRoutes.js.
router.put('/:id', bookController.updateBook);

// 删除图书 routes/bookRoutes.js
router.delete('/:id', bookController.deleteBook);

module.exports = router;

File: library-backend/routes/bookRoutes.go
package routes

import (
	"library-backend/controllers"
	"net/http"

	"github.com/gorilla/mux"
)

// BookRoutes 设置图书相关的路由 / Set book-related routes
func BookRoutes(router *mux.Router) {

	router.HandleFunc("/books", controllers.GetAllBooks).Methods(http.MethodGet)
	router.HandleFunc("/books/{id:[0-9]+}", controllers.GetBookById).Methods(http.MethodGet)
	router.HandleFunc("/books", controllers.AddBook).Methods(http.MethodPost)
	router.HandleFunc("/books/{id:[0-9]+}", controllers.UpdateBook).Methods(http.MethodPut)
	router.HandleFunc("/books/{id:[0-9]+}", controllers.DeleteBook).Methods(http.MethodDelete)
}
